packages/@react-spectrum/s2/style/style-macro.ts

```diff
@@ -76,6 +76,22 @@ function mapConditionalValue<T, U>(value: PropertyValueDefinition<T>, fn: (value
   }
 }
 
+function mapConditionalShorthand<T, C extends string, R extends RenderProps<string>>(value: PropertyValueDefinition<T>, fn: ShorthandProperty<T>): {[property: string]: StyleValue<Value, C, R>} {
+  if (typeof value === 'object') {
+    let res = {};
+    for (let condition in value) {
+      let properties = mapConditionalShorthand(value[condition], fn);
+      for (let property in properties) {
+        res[property] ??= {};
+        res[property][condition] = properties[property];
+      }
+    }
+    return res;
+  } else {
+    return fn(value);
+  }
+}
```

Comment:
This enables conditional values in shorthand properties, e.g. something like this:

```js
style({
  font: {
    size: {
      S: 'ui-xs',
      M: 'ui-sm',
      L: 'ui',
      XL: 'ui-lg'
    }
  }
})
```

That will now expand to individual properties correctly.